src/Datadog.Trace/Logging/LogRateLimiter.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Concurrent;
+
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Logging
+{
+    internal class LogRateLimiter : ILogRateLimiter
+    {
+#if NET45
+        private static readonly DateTime _unixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+#endif
+
+        private readonly int _secondsBetweenLogs;
+        private readonly ConcurrentDictionary<LogRateBucketKey, LogRateBucketInfo> _buckets
+            = new ConcurrentDictionary<LogRateBucketKey, LogRateBucketInfo>();
+
+        public LogRateLimiter(int secondsBetweenLogs)
+        {
+            if (secondsBetweenLogs < 1)
+            {
+                throw new ArgumentException("Must have positive number of seconds between logs", nameof(secondsBetweenLogs));
+            }
+
+            _secondsBetweenLogs = secondsBetweenLogs;
+        }
+
+        /// <inheritdoc/>
+        public bool ShouldLog(string filePath, int lineNumber, out uint skipCount)
+        {
+            if (filePath == string.Empty && lineNumber == 0)
```

Comment:
Following up on this, should we do an OR statement, instead of the AND statement?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1153/commits/56e9189e9e4430d2485401d2031f54f3204a35ca